### PR TITLE
hack/usco23: Introduce section 05: Debugging

### PR DESCRIPTION
**Pull request by vladandrew**
This pull request introduces the fifth session about debugging under Unikraft. This session goes over common debugging techniques and how we can use them on Unikraft.

### CODE REVIEW

File: content/en/community/hackathons/usoc23/debugging/index.md [hunk 1]
```diff
@@ -0,0 +1,92 @@
+---
+title: Session 05 - Debugging in Unikraft
+date: 2023-04-13T05:33:37+10:00
+weight: 4
```
Comment (StefanJum) [inline content/en/community/hackathons/usoc23/debugging/index.md:4]:
```suggestion
weight: 5
```

File: content/en/community/hackathons/sessions/debugging/content/Debugging_Using-GDB.md [hunk 2]
```diff
@@ -8,6 +8,10 @@ We recommend 3, the highest level.
 
 ![debug information level](/community/hackathons/sessions/debugging/images/debug_information_level.png)
 
+In most cases, we'll be running our unikernel using QEMU. We won't be able to simply run the binary
+with GDB, but we be using [gdb’s remote-connection facility](https://qemu-project.gitlab.io/qemu/system/gdb.html).
+
+
```
Comment (StefanJum) [inline content/en/community/hackathons/sessions/debugging/content/Debugging_Using-GDB.md:14]:
```suggestion

```

**Review by StefanJum — CHANGES_REQUESTED** [review]
@vladandrew I know this is WIP, just a few comments.

File: content/en/community/hackathons/usoc23/debugging/index.md [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+
+### 07. Bonus. Bad ELF in Town
+
+{{< readfile file="/community/hackathons/sessions/debugging/content/Practical-Work_07.-Bonus.-Bad-ELF-in-Town.md" markdown="true" >}}
+
+### 08. Give Us Feedback
+
+We want to know how to make the next sessions better.
+For this, we need your [feedback](https://docs.google.com/forms/d/e/1FAIpQLSdpyEpCW0yxJBSDa7Iq95JUCe5YNnRpxOD31ol4ezpKWnqrhA/viewform).
+Thank you!
```
Comment (StefanJum) [inline content/en/community/hackathons/usoc23/debugging/index.md:88]: Remove this, it's for last year.

File: content/en/community/hackathons/usoc23/debugging/index.md [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+
+### 03. Bug or feature?
+
+{{< readfile file="/community/hackathons/sessions/debugging/content/Practical-Work_03.-Bug-or-feature?.md" markdown="true" >}}
+
+### 04. Nginx with or without main? That's the question.
+
+{{< readfile file="/community/hackathons/sessions/debugging/content/Practical-Work_06.-Nginx-with-or-without-main?-That's-the-question..md" markdown="true" >}}
+
+### 05. Bad ELF in Town
```
Comment (StefanJum) [inline content/en/community/hackathons/usoc23/debugging/index.md:48]: @vladandrew Please also remove this, it's too much for the scope of the session

File: content/en/community/hackathons/usoc23/debugging/index.md [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+
+{{< readfile file="/community/hackathons/sessions/debugging/content/Practical-Work_04.-Tutorial.-Use-Tracepoints..md" markdown="true" >}}
+
+### 05. Can you trace your own program?
+
+{{< readfile file="/community/hackathons/sessions/debugging/content/Practical-Work_05.-Can-you-trace-your-own-program?.md" markdown="true" >}}
+
+### 06. Nginx with or without main? That's the question.
+
+{{< readfile file="/community/hackathons/sessions/debugging/content/Practical-Work_06.-Nginx-with-or-without-main?-That's-the-question..md" markdown="true" >}}
```
[the remark this answers is not in the document]
Comment (vladandrew) [reply content/en/community/hackathons/usoc23/debugging/index.md:46]: Done

File: content/en/community/hackathons/usoc23/debugging/index.md [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+
+## Practical Work
+
+### Support Files
+
+{{< readfile file="/community/hackathons/sessions/debugging/content/Practical-Work_Support-Files.md" markdown="true" >}}
+
+### 01. Tutorial. Use GDB in Unikraft
+
+{{< readfile file="/community/hackathons/sessions/debugging/content/Practical-Work_01.-Tutorial.-Use-GDB-in-Unikraft.md" markdown="true" >}}
```
[the remark this answers is not in the document]
Comment (vladandrew) [reply content/en/community/hackathons/usoc23/debugging/index.md:34]: Done for all

File: content/en/community/hackathons/sessions/debugging/content/Practical-Work_01.-Tutorial.-Use-GDB-in-Unikraft.md [hunk 7]
```diff
@@ -4,32 +4,37 @@ First make sure you have the following conventional working directory also shown
 
 ```
```
Comment (StefanJum) [inline content/en/community/hackathons/sessions/debugging/content/Practical-Work_01.-Tutorial.-Use-GDB-in-Unikraft.md:5]:
```suggestion
```text
```

File: content/en/community/hackathons/sessions/debugging/content/Practical-Work_01.-Tutorial.-Use-GDB-in-Unikraft.md [hunk 8]
```diff
@@ -4,32 +4,37 @@ First make sure you have the following conventional working directory also shown
...
-|-- libs/
-`-- unikraft/
+app-helloworld/
+├── Makefile
+....
+└── .unikraft
+    └── unikraft
+    └── libs
+        └── lib-...
+
```
Comment (StefanJum) [inline content/en/community/hackathons/sessions/debugging/content/Practical-Work_01.-Tutorial.-Use-GDB-in-Unikraft.md:14]:
```suggestion
app-helloworld/
|-- Makefile
....
`-- .unikraft
    `-- unikraft
    `-- libs
        `-- lib-...

```
Use ascii output

**Review by StefanJum — CHANGES_REQUESTED** [review]
Just few more minor comments, we can merge it after that @vladandrew

**Review by StefanJum — APPROVED** [review]
All good now, thanks @vladandrew 
Reviewed-by: Stefan Jumarea <stefanjumarea02@gmail.com>
Approved-by: Stefan Jumarea <stefanjumarea02@gmail.com>

**Pull request merged**